pkg/crc/cluster/cluster.go

```diff
@@ -22,21 +22,44 @@ func WaitForSsh(driver drivers.Driver) error {
 	return errors.RetryAfter(10, checkSshConnectivity, time.Second)
 }
 
-// CheckCertsValidity checks if the cluster certs have expired
-func CheckCertsValidity(driver drivers.Driver) error {
+// CheckCertsValidityUsingBundleBuildTime check if the cluster certs going to expire soon.
+func CheckCertsValidityUsingBundleBuildTime(buildTime time.Time) (bool, int) {
+	certExpiryDate := buildTime.AddDate(0, 1, 0)
+	// Warn user if the cert expiry going to happen starting of the 7 days
+	timeAfter7Days := time.Now().AddDate(0, 0, 20)
```

Comment:
@praveenkumar can you take a look at this again, I think you meant to add 7 days!

Reply:
@anjannath Changed, Good catch :+1: